### ISSUE TITLE
`forth`: `evaluateBoth` property

### ISSUE THREAD

**Issue by NobbZ** [exercism/problem-specifications#2018]
In the `forth` specification there is a single test for the `evaluateBoth` property.

The remaining tests are for the `evaluate` property.

https://github.com/exercism/problem-specifications/blob/main/exercises/forth/canonical-data.json#L510

This confused the erlang test generator recently, as it assumes the `property` key points to the function under test.

Is this a missunderstanding on my side or do we need to find another way to specify a sequence of operations/function calls in the spec?

**Comment by kytrinyx** [comment]
I think a `scenario` is a very good suggestion.
How about `local-scope`? We already have that scenario, and it seems to fit the bill.

**Comment by ErikSchierboom** [comment]
> How about `local-scope`? We already have that scenario, and it seems to fit the bill.

Yes, that does indeed fit the bill well and should indeed be added!

What I was originally suggesting was to add _another_ scenario, namely one that indicates that the property is different from a regular property and requires special treatment by a test generator. But maybe we don't need it as certain scopes will always require this special treatment, like `local-scope` and `immutable`. Maybe it is enough to identify all test cases that have special behavior and a add an appropriate scenario?

Here is another weird test case:

```json
{
  "uuid": "2ca77b9b-c099-46c3-a02c-0d0f68ffa0fe",
  "description": "each ability is only calculated once",
  "property": "strength",
  "input": {},
  "expected": "strength == strength"
}
```

Should this have a `memoized` or `memoization` scenario?

**Comment by kytrinyx** [comment]
Ah, yeah, `memoization` is probably a good idea there.
